### DISCUSSION TITLE
Add Filter and Sort  to more modules

### DISCUSSION THREAD

**Discussion by maximo1984** [Kareadita/Kavita#2890] [Ideas]
### Idea Description

Idea - Add Filter and Sort to the Collections, Reading Lists, and Volume pages. This will allow more power in searching for specific comic books by refining search group to specific sets. This would also allow pulling single issues into the search results rather than a set of volumes. Example, I just searched for an artist and the result was Action Comics (which has all volumes in it). So searching for the artist is useless here.

### Idea Category

API

### Duration of Using Kavita

3 months

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
I need more information on what you are expecting for each?

Like how would you sort and filter reading lists, etc. Otherwise this will probably never get implemented.

**Comment by maximo1984** [comment]
Currently, there is no way to find individual issues by favorite creators. For example here, searching for "Greg Capullo", I only get the Batman volumes as they were imported. There are over 800 issues in there. The search feature IS finding the comics that he drew, so you already have that data, why not display the individual issues? And once we get that working, there needs to be an action to add individual issues TO collections. I want to make collections of my favorite artists (one example of why this is needed).

![Screenshot from 2024-04-20 08-03-24](https://github.com/Kareadita/Kavita/assets/40921347/f644d213-8e2a-4fe7-9738-f5fc52f24f1e)
 The filter and search can search in "Libraries", but I think there should also be an option to search for single issues inside Reading Lists and Collections.

**Comment by maximo1984** [comment]
Any thoughts on adding individual issues and reading lists to collections? 

I have a little bit of coding experience. I could probably figure it out. What is the best way to get started editing the main code for local use?

Yet another example of use for adding reading lists to collections is this screenshot of TPB. You can see there are some with same themes art style (new 52). This theme is consistent across all TPB for New 52. I'd like to add each of these TPB to a new Collection called "New 52"
![Screenshot from 2024-04-27 17-54-39](https://github.com/Kareadita/Kavita/assets/40921347/7f13a456-1df2-485e-a294-f3c6e0de9264)

**Comment by maximo1984** [comment]
Just to hammer this home, what good are collections if you cannot make a collection of single issues with characters, comic eras, writers, artists, etc?

**Reply by majora2007** [reply]
Collections in other apps are all series based. I use collections to represent themes, best by year, my favorites, etc. 

I don't think collections are the best place for what you're wanting. You wan an unordered list of issues with some theme it seems. Or perhaps a filtering system on issues rather than series (something only useful for books and comics).
